### PR TITLE
RD2: New Closed Reason field

### PR DESCRIPTION
**Pull request by force2b**
# Critical Changes

# Changes

- Added a new Closed Reason picklist field to the Recurring Donation object. This field is not associated with any specific automation and can be added to the existing Recurring Donation page layout in customer orgs if wanted.

# Issues Closed

# New Metadata

- npe03__Recurring_Donation__c.ClosedReason__c

# Deleted Metadata

### CODE REVIEW

**Comment by force2b** [comment]
**lurch: attach W-037453

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HoUvQAK" target="_blank">W-037453</a>

File: unpackaged/config/rd2_post_config/layouts/npe03__Recurring_Donation__c-Enhanced Recurring Donations Layout.layout [hunk 1]
```diff
@@ -42,6 +42,10 @@
                 <behavior>Edit</behavior>
                 <field>npe03__Recurring_Donation_Campaign__c</field>
             </layoutItems>
+            <layoutItems>
+                <behavior>Edit</behavior>
+                <field>%%%NAMESPACE%%%ClosedReason__c</field>
```
Comment (npsp-reedestockton) [inline unpackaged/config/rd2_post_config/layouts/npe03__Recurring_Donation__c-Enhanced Recurring Donations Layout.layout]: I learned something new about cci.  Cool!

**Comment by force2b** [comment]
@rponti-sforg This is ready for testing.

**Comment by rponti-sforg** [comment]
QE testing is complete for this PR.  Tests have passed.

For further testing details, see:
https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x1E000001HoUvQAK/view
https://salesforce.quip.com/ROUmAnsqZEfq

Will merge PR once builds are finished running and set WI status to Closed.

**Pull request merged**